python/pyiceberg/expression/__init__.py

```diff
@@ -0,0 +1,83 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from pyiceberg.expression.expressions import (
+    And,
+    Equal,
+    GreaterThan,
+    GreaterThanEqual,
+    In,
+    IsNaN,
+    IsNull,
+    LessThan,
+    LessThanEqual,
+    Not,
+    NotEqual,
+    NotIn,
+    NotNaN,
+    NotNull,
+    Or,
+)
+from pyiceberg.expression.literals import (
+    AboveMax,
+    BelowMin,
+    BinaryLiteral,
+    BooleanLiteral,
+    DateLiteral,
+    DecimalLiteral,
+    DoubleLiteral,
+    FixedLiteral,
+    FloatLiteral,
+    LongLiteral,
+    StringLiteral,
+    TimeLiteral,
+    TimestampLiteral,
+    UUIDLiteral,
```

Comment:
Do we need the literal classes? I think the `literal` method should be enough. And we should also coerce values in predicates.